### ISSUE TITLE
Create uncertainty mixin and SpectralCube with uncertainty class

### ISSUE THREAD

**Issue by brechmos** [radio-astro-tools/spectral-cube#574]
Based on the discussion in https://github.com/radio-astro-tools/spectral-cube/issues/572, there is a desire to have a SpectralCubeWithUncertainty and an uncertainty-attribute-containing mixin.

**Comment by brechmos** [comment]
There are a couple of options: 1) the uncertainty is a numpy array (or something like that) but then the spectral cube class or the user would have to remember what type it is and/or make assumptions that it is a standard deviation or 2) leverage the uncertainties in astropy.nddata and force the user to state the type of uncertainty it is.

By forcing the user to tell us the type of uncertainty then higher level functions (for example) don't need to make assumptions about the type of uncertainty we have, but would know what it is which would calculations easier. 

(I guess it comes down to: at some point we are going to have to know the type of uncertainty, and I would lean to force the issue up front.)

**Comment by keflavich** [comment]
Right, I meant that instead of a user specifying `uncertainty=(1, 'uncertainty_type')`, they'd specify `uncertainty=(1, UncertaintyClass)`.   However, I don't like that callspec much - it means the user has to construct an appropriately structured tuple with no UI-provided clues.  Something like `uncertainty_extension=1, uncertainty_type=UncertaintyClass` would be better.

**Comment by brechmos** [comment]
Sorry, I didn't mean they would specify an astropy class/type, just a string. I'll mock it up and create a PR for perusal and further discussion.
